core/src/main/scala/kafka/server/KafkaServer.scala

```diff
@@ -547,6 +547,8 @@ class KafkaServer(val config: KafkaConfig, time: Time = Time.SYSTEM, threadNameP
 
         if (metrics != null)
           CoreUtils.swallow(metrics.close())
+        if (socketServer != null)
+          CoreUtils.swallow(socketServer.closeRequestMetrics())
```

Comment:
Would it make sense to close this in `KafkaRequestHandlerPool`?

Reply:
@ijuma Thank you for the review, updated.